File: frontend/src/types/notification.ts
// 告警通知类型

export type NotificationLevel = 'critical' | 'warning' | 'info'

export interface NotificationRecord {
  id: string
  level: NotificationLevel
  event_type: string
  title: string
  message: string
  details: Record<string, unknown>
  run_id?: string
  status: string
  is_read: boolean
  created_at: string
  updated_at: string
}

export interface UnreadCount {
  count: number
}

File: src/squant/schemas/notification.py
"""Notification schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class NotificationResponse(BaseModel):
    """Notification record response."""

    id: str
    level: str
    event_type: str
    title: str
    message: str
    details: dict[str, Any]
    run_id: str | None
    status: str
    is_read: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class UnreadCountResponse(BaseModel):
    """Unread notification count."""

    count: int


class MarkReadRequest(BaseModel):
    """Mark notifications as read."""

    notification_ids: list[str] | None = Field(
        default=None,
        description="Notification IDs to mark as read. None = mark all as read.",
    )
